libs/coroutines/include/hpx/coroutines/detail/context_base.hpp

```diff
@@ -204,6 +214,11 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
             delete_tss_storage(m_thread_data);
 #else
             m_thread_data = 0;
+#ifdef HPX_HAVE_LIBCDS
+            libcds_data_ = 0;
+            libcds_hazard_pointer_data_ = 0;
+            libcds_dynamic_hazard_pointer_data_ = 0;
+#endif
```

Comment:
I think this #if/#endif should be outside the `HPX_HAVE_THREAD_LOCAL_STORAGE` condition above.